### PR TITLE
DPC-588 Feature: Changed successful token deletion response from 200 to 204.

### PR DESCRIPTION
**Pull request by SalGutierrez**
### Fixes [DPC-588](https://jira.cms.gov/browse/DPC-588)

Delete /Token/{id} endpoint returns a 200 with an empty body, outcome is not clear if it was successful or not.

### Proposed Changes

Original proposal was to include a body with a successful response, but after consulting with team we decided it was best to return a 204 (No Content) status code and an empty body.

### Change Details

Delete /Token/{id} endpoint now returns a 204 when a token is successfully deleted.
Update TokenResource and unit tests.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Yes, via integration test.

### CODE REVIEW

File: dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/TokenResourceTest.java [hunk 1]
```diff
@@ -152,7 +152,8 @@ void testTokenDeletion() throws IOException {
             final HttpDelete httpDelete = new HttpDelete(getBaseURL() + String.format("/Token/%s", token.getId()));
             httpDelete.addHeader(HttpHeaders.AUTHORIZATION, String.format("Bearer %s", this.fullyAuthedToken));
             try (CloseableHttpResponse response = client.execute(httpDelete)) {
-                assertEquals(HttpStatus.OK_200, response.getStatusLine().getStatusCode(), "Should have succeeded");
+                assertEquals(HttpStatus.NO_CONTENT_204, response.getStatusLine().getStatusCode(), "Should have succeeded with a 204 (No Content)");
```
Comment (dhgreene) [inline dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/TokenResourceTest.java:155]: @Sun-Mountain Will this require a coordinated change to [APIClient.response_successful?](https://github.com/CMSgov/dpc-app/blob/557bf1381d2c615ba15f57a8c0aee89b2a5b4539/dpc-web/app/services/api_client.rb#L74) and perhaps other changes?

Comment (Sun-Mountain) [reply dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/TokenResourceTest.java:155]: Yes. At least in few places.

Comment (Sun-Mountain) [reply dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/TokenResourceTest.java:155]: Should I make a ticket and start working on it?

Comment (Sun-Mountain) [reply dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/TokenResourceTest.java:155]: 🖖

**Pull request merged**